lib/funnel_web/controllers/repositories_controller.ex

```diff
@@ -6,49 +6,76 @@ defmodule FunnelWeb.RepositoriesController do
   alias Funnel.Repo
 
   plug :scrub_params, "repository" when action in [:update]
+  plug :git_hub_authenticate
+  plug :git_hub_authorize_repository when action in [:show, :edit, :update]
 
   def index(conn, _params) do
-    alias FunnelWeb.Router.Helpers
-    case get_session(conn, :git_hub_access_token) do
-      nil ->
-        redirect(conn, to: Helpers.auth_path(conn, :login))
-      token ->
-        repositories = Platter.get_all_user_repositories(token)
-        render(conn, "index.html", repositories: repositories)
-    end
+    repositories = Platter.get_all_user_repositories(conn.assigns[:git_hub_access_token])
+    render(conn, "index.html", repositories: repositories)
   end
 
-  def show(conn, %{"id" => id}) do
-    repository = Repo.preload(GitHub.get_repository!(id), :strategy)
-    render(conn, "show.html", repository: repository)
+  def show(conn, _params) do
+    render(conn, "show.html")
   end
 
-  def edit(conn, %{"id" => id}) do
-    repository = Repo.preload(GitHub.get_repository!(id), :strategy)
+  def edit(conn, _params) do
     strategies = Git.list_strategies()
-    changeset = GitHub.change_repository(repository)
+    changeset = GitHub.change_repository(conn.assigns[:repository])
     render(
       conn,
       "edit.html",
-      [repository: repository, strategies: strategies, changeset: changeset]
+      [strategies: strategies, changeset: changeset]
     )
   end
 
-  def update(conn, %{"id" => id, "repository" => repository_params}) do
-    repository = Repo.preload(GitHub.get_repository!(id), :strategy)
-    strategies = Git.list_strategies()
-
-    case GitHub.update_repository(repository, repository_params) do
+  def update(conn, %{"repository" => repository_params}) do
+    case GitHub.update_repository(conn.assigns[:repository], repository_params) do
       {:ok, repository} ->
         conn
+        |> assign(:repository, repository)
         |> put_flash(:info, "Repository updated successfully.")
         |> redirect(to: repositories_path(conn, :show, repository))
       {:error, %Ecto.Changeset{} = changeset} ->
+        strategies = Git.list_strategies()
+        repository = conn.assigns[:repository]
         render(
         conn,
         "edit.html",
         [repository: repository, strategies: strategies, changeset: changeset]
         )
     end
   end
+
+  defp git_hub_authenticate(conn, _) do
+    alias FunnelWeb.Router.Helpers
+    case get_session(conn, :git_hub_access_token) do
```

Comment:
how does this session persist?

Reply:
it looks like the default is to store this encrypted in the cookie itself. if i'm reading this right

http://phoenixframework.org/blog/sessions
https://hexdocs.pm/plug/Plug.Session.COOKIE.html#content